src/main/java/edu/cornell/kfs/fp/service/impl/RecurringDisbursementVoucherDocumentServiceImpl.java

```diff
@@ -409,8 +417,137 @@ public Object call() throws WorkflowException {
             throw new RuntimeException("cancelDVAsSystemUser() Unable to cancel DV: " + dvDocumentNumber, e);
         }
     }
+
+    @Override
+    public boolean autoApproveDisbursementVouchersSpawnedByRecurringDvs() {
+        LOG.info("autoApproveDisbursementVouchersSpawnedByRecurringDvs: Entered.");
+        int approvalCount = 0;
+        int errorCount = 0;
+        List <String> dvDocIdCausingError = new ArrayList<String>();
+        Collection <String> dvDocIds = getRecurringDisbursementVoucherSearchDao().findSavedDvIdsSpawnedByRecurringDvForCurrentAndPastFiscalPeriods(getCurrentFiscalPeriodEndDate());
+        if (ObjectUtils.isNotNull(dvDocIds) && !(dvDocIds.isEmpty())) {
+            for (String dvDocId :  dvDocIds) {
+                LOG.info("autoApproveDisbursementVouchersSpawnedByRecurringDvs: Processing start for DV ID:: " + dvDocId);
+                CuDisbursementVoucherDocument dv = null;
+                try{
+                    dv = getDisbursementVoucher(dvDocId);
+                } catch (WorkflowException e) {
```

Comment:
The workflow exception has already been logged by the function that was called. I really don't think we want to double log them (Do we Jay?)  In this routine, we have the control logic for the batch processing so the number of errors and the actual doc # that error occurred on can be tracked and reported back to the user at the end of this method via output to the log file.  I think adding another log entry here would be just noise. (Jay: your take?)